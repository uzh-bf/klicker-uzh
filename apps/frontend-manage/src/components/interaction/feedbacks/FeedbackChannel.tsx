// TODO: notifications

import { faEye, faEyeSlash } from '@fortawesome/free-regular-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  Feedback as FeedbackType,
  FeedbackResponse,
} from '@klicker-uzh/graphql/dist/ops'
import { Button } from '@uzh-bf/design-system'

import useFeedbackFilter from '../../../lib/hooks/useFeedbackFilter'
// import { createNotification, requestNotificationPermissions } from '../../../lib/utils/notifications'
import Feedback from './Feedback'
import FeedbackSearchAndFilters from './FeedbackSearchAndFilters'

interface Props {
  feedbacks?: FeedbackType[]
  handleDeleteFeedback: (feedbackId: number) => void
  handlePinFeedback: (feedbackId: number, isPinned: boolean) => void
  handlePublishFeedback: (feedbackId: number, isPublished: boolean) => void
  handleResolveFeedback: (feedbackId: number, resolvedState: boolean) => void
  handleRespondToFeedback: (feedbackId: number, response: string) => void
  handleDeleteFeedbackResponse: (responseId: number) => void
  isActive?: boolean
  isPublic?: boolean
}

const defaultProps = {
  feedbacks: [],
  isActive: false,
  isPublic: false,
}

function FeedbackChannel({
  feedbacks,
  isPublic,
  handleDeleteFeedback,
  handlePinFeedback,
  handlePublishFeedback,
  handleResolveFeedback,
  handleRespondToFeedback,
  handleDeleteFeedbackResponse,
}: Props) {
  const [sortedFeedbacks, filterProps] = useFeedbackFilter(feedbacks, {
    withSearch: true,
  })
  // const [feedbackLength, setFeedbackLength] = useState(0)

  // useEffect(() => {
  //   requestNotificationPermissions((permission) => {
  //     if (permission === 'granted') {
  //       setFeedbackLength(feedbacks.length)
  //     }
  //   })
  // }, [])

  // useEffect(() => {
  //   if (!sessionStorage?.getItem(`feedback ${feedbacks[feedbacks.length - 1]?.id}`)) {
  //     if (feedbacks.length > feedbackLength) {
  //       createNotification(intl.formatMessage(messages.notificationTitle), feedbacks[feedbacks.length - 1].content)
  //     }
  //     sessionStorage?.setItem(`feedback ${feedbacks[feedbacks.length - 1]?.id}`, 'notified')
  //   }
  //   setFeedbackLength(feedbacks.length)
  // }, [feedbacks.length])

  return (
    <div>
      <FeedbackSearchAndFilters
        disabled={sortedFeedbacks?.length === 0}
        {...filterProps}
      />
      <div className="flex flex-col gap-2 mt-4 overflow-y-auto">
        {/* // TODO: styling */}
        {!feedbacks ||
          (feedbacks.length === 0 && (
            <div>Bisher keine Feedbacks erhalten...</div>
          ))}

        {/* // TODO: styling */}
        {feedbacks &&
          feedbacks.length > 0 &&
          sortedFeedbacks &&
          sortedFeedbacks.length === 0 && (
            <div>
              Keine Feedbacks stimmen mit den aktuellen Filtereinstellungen
              Ã¼berein...
            </div>
          )}

        {sortedFeedbacks?.map(
          ({
            id,
            content,
            createdAt,
            votes,
            isResolved,
            isPinned,
            isPublished,
            responses,
            resolvedAt,
          }: FeedbackType) => (
            <div className="flex flex-row print:mt-2" key={id}>
              {!isPublic && (
                <div className="flex-initial print:hidden">
                  <Button
                    className="justify-center mr-2 w-9 h-9"
                    onClick={() => {
                      handlePublishFeedback(id, !isPublished)
                    }}
                  >
                    {isPublished ? (
                      <Button.Icon>
                        <FontAwesomeIcon icon={faEye} />
                      </Button.Icon>
                    ) : (
                      <Button.Icon>
                        <FontAwesomeIcon icon={faEyeSlash} />
                      </Button.Icon>
                    )}
                  </Button>
                </div>
              )}
              <div className="flex-1">
                <Feedback
                  id={id}
                  content={content}
                  createdAt={createdAt}
                  pinned={isPinned}
                  resolved={isResolved}
                  resolvedAt={resolvedAt}
                  responses={responses as FeedbackResponse[]}
                  votes={votes}
                  onDeleteFeedback={() => handleDeleteFeedback(id)}
                  onDeleteResponse={(responseId) =>
                    handleDeleteFeedbackResponse(responseId)
                  }
                  onPinFeedback={(pinState) => handlePinFeedback(id, pinState)}
                  onResolveFeedback={(resolvedState) =>
                    handleResolveFeedback(id, resolvedState)
                  }
                  onRespondToFeedback={(id, response) =>
                    handleRespondToFeedback(id, response)
                  }
                />
              </div>
            </div>
          )
        )}

        {/* // TODO: still include this banner?
        {hasSurveyBannerInitialized && (isSurveyBannerVisible ?? true) && (
          <div className="mt-2 print:hidden">
            <Message
              warning
              content={
                <FormattedMessage
                  defaultMessage="If you have used our feedback-channel (Q&A) functionality, please consider participating in our 2-minute survey under this {link}."
                  id="runningSession.audienceInteraction.survey"
                  values={{
                    link: (
                      <a href="https://hi.switchy.io/6IeK" rel="noreferrer" target="_blank">
                        link
                      </a>
                    ),
                  }}
                />
              }
              icon="bullhorn"
              size="tiny"
              onDismiss={() => setIsSurveyBannerVisible(false)}
            />
          </div>
        )} */}
      </div>
    </div>
  )
}

FeedbackChannel.defaultProps = defaultProps

export default FeedbackChannel
