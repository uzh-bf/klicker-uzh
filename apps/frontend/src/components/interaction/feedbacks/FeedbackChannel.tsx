import { faEye, faEyeSlash } from '@fortawesome/free-regular-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Button } from '@uzh-bf/design-system'
import { useEffect, useState } from 'react'
import { defineMessages, FormattedMessage, useIntl } from 'react-intl'
import { Message } from 'semantic-ui-react'

import useFeedbackFilter from '../../../lib/hooks/useFeedbackFilter'
import useStickyState from '../../../lib/hooks/useStickyState'
import { createNotification, requestNotificationPermissions } from '../../../lib/utils/notifications'
import Feedback from './Feedback'
import FeedbackSearchAndFilters from './FeedbackSearchAndFilters'

const messages = defineMessages({
  notificationTitle: {
    defaultMessage: 'New Question / Feedback',
    id: 'feedbackchannel.notifications.title',
  },
})

interface Props {
  feedbacks?: any[]
  handleDeleteFeedback: any
  handlePinFeedback: (id: string, pinState: boolean) => void
  handlePublishFeedback: (id: string, publishState: boolean) => void
  handleResolveFeedback: (id: string, resolvedState: boolean) => void
  handleRespondToFeedback: (id: string, response: string) => void
  handleDeleteFeedbackResponse: (id: string, responseId: string) => void
  isActive?: boolean
  isPublic?: boolean
}

const defaultProps = {
  feedbacks: [],
  isActive: false,
  isPublic: false,
}

function FeedbackChannel({
  feedbacks,
  isPublic,
  handleDeleteFeedback,
  handlePinFeedback,
  handlePublishFeedback,
  handleResolveFeedback,
  handleRespondToFeedback,
  handleDeleteFeedbackResponse,
}: Props) {
  const [isSurveyBannerVisible, setIsSurveyBannerVisible, hasSurveyBannerInitialized] = useStickyState(
    true,
    'qa-survey-lecturer-visible'
  )

  const [sortedFeedbacks, filterProps] = useFeedbackFilter(feedbacks, { withSearch: true })
  const [feedbackLength, setFeedbackLength] = useState(0)
  const intl = useIntl()

  useEffect(() => {
    requestNotificationPermissions((permission) => {
      if (permission === 'granted') {
        setFeedbackLength(feedbacks.length)
      }
    })
  }, [])

  useEffect(() => {
    if (!sessionStorage?.getItem(`feedback ${feedbacks[feedbacks.length - 1]?.id}`)) {
      if (feedbacks.length > feedbackLength) {
        createNotification(intl.formatMessage(messages.notificationTitle), feedbacks[feedbacks.length - 1].content)
      }
      sessionStorage?.setItem(`feedback ${feedbacks[feedbacks.length - 1]?.id}`, 'notified')
    }
    setFeedbackLength(feedbacks.length)
  }, [feedbacks.length])

  return (
    <div>
      <FeedbackSearchAndFilters disabled={sortedFeedbacks?.length === 0} {...filterProps} />

      <div className="flex flex-col gap-2 mt-4 overflow-y-auto">
        {feedbacks.length === 0 && (
          <Message info>
            <FormattedMessage defaultMessage="No feedbacks received yet..." id="runningSession.info.nofeedbacks" />
          </Message>
        )}

        {feedbacks.length > 0 && sortedFeedbacks.length === 0 && (
          <Message info>
            <FormattedMessage
              defaultMessage="No feedbacks matching your current filter selection..."
              id="runningSession.info.nomatchingfeedbacks"
            />
          </Message>
        )}

        {sortedFeedbacks.map(
          ({ id, content, createdAt, votes, resolved, pinned, published, responses, resolvedAt }) => (
            <div className="flex flex-row print:mt-2" key={id}>
              {!isPublic && (
                <div className="flex-initial print:hidden">
                  <Button
                    className="justify-center mr-2 w-9 h-9"
                    onClick={() => {
                      handlePublishFeedback(id, !published)
                    }}
                  >
                    {published ? (
                      <Button.Icon>
                        <FontAwesomeIcon icon={faEye} />
                      </Button.Icon>
                    ) : (
                      <Button.Icon>
                        <FontAwesomeIcon icon={faEyeSlash} />
                      </Button.Icon>
                    )}
                  </Button>
                </div>
              )}
              <div className="flex-1">
                <Feedback
                  content={content}
                  createdAt={createdAt}
                  pinned={pinned}
                  resolved={resolved}
                  resolvedAt={resolvedAt}
                  responses={responses}
                  votes={votes}
                  onDeleteFeedback={() => handleDeleteFeedback(id)}
                  onDeleteResponse={(responseId) => handleDeleteFeedbackResponse(id, responseId)}
                  onPinFeedback={(pinState) => handlePinFeedback(id, pinState)}
                  onResolveFeedback={(resolvedState) => handleResolveFeedback(id, resolvedState)}
                  onRespondToFeedback={(response) => handleRespondToFeedback(id, response)}
                />
              </div>
            </div>
          )
        )}

        {hasSurveyBannerInitialized && (isSurveyBannerVisible ?? true) && (
          <div className="mt-2 print:hidden">
            <Message
              warning
              content={
                <FormattedMessage
                  defaultMessage="If you have used our feedback-channel (Q&A) functionality, please consider participating in our 2-minute survey under this {link}."
                  id="runningSession.audienceInteraction.survey"
                  values={{
                    link: (
                      <a href="https://hi.switchy.io/6IeK" rel="noreferrer" target="_blank">
                        link
                      </a>
                    ),
                  }}
                />
              }
              icon="bullhorn"
              size="tiny"
              onDismiss={() => setIsSurveyBannerVisible(false)}
            />
          </div>
        )}
      </div>
    </div>
  )
}

FeedbackChannel.defaultProps = defaultProps

export default FeedbackChannel
