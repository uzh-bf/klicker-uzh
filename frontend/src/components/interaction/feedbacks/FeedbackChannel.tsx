import React, { useEffect } from 'react'
import { Checkbox } from 'semantic-ui-react'
import { defineMessages, FormattedMessage, useIntl } from 'react-intl'

import Feedback from './Feedback'

const messages = defineMessages({
  activated: {
    defaultMessage: 'Activated',
    id: 'common.string.activated',
  },
  publishQuestions: {
    defaultMessage: 'Publish feedbacks',
    id: 'runningSession.feedbackChannel.publishFeedbacks',
  },
})

interface Props {
  feedbacks?: any[]
  handleActiveToggle: any
  handleDeleteFeedback: any
  handlePublicToggle: any
  isActive?: boolean
  isPublic?: boolean
  subscribeToMore: any
}

const defaultProps = {
  feedbacks: [],
  isActive: false,
  isPublic: false,
}

function FeedbackChannel({
  feedbacks,
  isActive,
  isPublic,
  handleActiveToggle,
  handlePublicToggle,
  handleDeleteFeedback,
  subscribeToMore,
}: Props): React.ReactElement {
  useEffect((): void => {
    if (subscribeToMore) {
      subscribeToMore()
    }
  }, [subscribeToMore])

  const intl = useIntl()

  return (
    <div>
      <div>
        <FormattedMessage
          defaultMessage="The Feedback-Channel allows you to get open feedback from your participants. If you publish the feedbacks, your participants will be able to see what other participants have already posted."
          id="runningSession.feedbackChannel.info"
        />
      </div>

      <div className="flex flex-col mt-4 md:flex-row">
        <div>
          <Checkbox
            toggle
            checked={isActive}
            defaultChecked={isActive}
            label={intl.formatMessage(messages.activated)}
            onChange={handleActiveToggle}
          />
        </div>
        <div className="mt-4 md:ml-8 md:mt-0">
          <Checkbox
            toggle
            checked={isPublic}
            defaultChecked={isPublic}
            disabled={!isActive}
            label={intl.formatMessage(messages.publishQuestions)}
            onChange={handlePublicToggle}
          />
        </div>
      </div>

      {isActive && (
        <div className="mt-4">
          {feedbacks.map(
            ({ id, content, votes }): React.ReactElement => (
              <div className="mt-2 first:mt-0" key={id}>
                <Feedback content={content} votes={votes} onDelete={(): void => handleDeleteFeedback(id)} />
              </div>
            )
          )}
        </div>
      )}
    </div>
  )
}

FeedbackChannel.defaultProps = defaultProps

export default FeedbackChannel
