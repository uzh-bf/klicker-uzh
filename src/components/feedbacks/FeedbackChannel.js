// @flow

import * as React from 'react'
import { Checkbox } from 'semantic-ui-react'
import { FormattedMessage } from 'react-intl'

import Feedback from './Feedback'

type Props = {
  data: Array<{
    content: string,
    id: string,
    votes: number,
  }>,
  intl: any,
  isActive?: boolean,
  isPublic?: boolean,
  handleActiveToggle: () => mixed,
  handlePublicToggle: () => mixed,
}

const defaultProps = {
  isActive: false,
  isPublic: false,
}

const FeedbackChannel = ({
  data,
  intl,
  isActive,
  isPublic,
  handleActiveToggle,
  handlePublicToggle,
}: Props) => (
  <div className="feedbackChannel">
    <h2>
      <FormattedMessage
        defaultMessage="Feedback-Channel"
        id="runningSession.feedbackChannel.string.title"
      />
    </h2>
    <div className="toggle">
      <Checkbox
        toggle
        defaultChecked={isActive}
        label={intl.formatMessage({
          defaultMessage: 'Activated',
          id: 'common.string.activated',
        })}
        value={isActive}
        onChange={handleActiveToggle}
      />
    </div>
    <div className="toggle publicationToggle">
      <Checkbox
        toggle
        className="publishCheckbox"
        defaultChecked={isPublic}
        disabled={!isActive}
        label={intl.formatMessage({
          defaultMessage: 'Publish questions',
          id: 'runningSession.feedbackChannel.string.publishQuestions',
        })}
        value={isPublic}
        onChange={handlePublicToggle}
      />
    </div>

    {isActive && (
      <div className="feedbacks">
        {data.map(({ content, id, votes }) => (
          <div className="feedback">
            <Feedback key={id} content={content} votes={votes} />
          </div>
        ))}
      </div>
    )}

    <style jsx>{`
      .feedbackChannel {
        display: flex;
        flex-direction: column;
      }

      h2,
      .toggle,
      .feedbacks {
        flex: 1;
      }

      h2 {
        margin-bottom: 1rem;
      }

      .publicationToggle {
        margin-top: 1rem;
      }

      .feedbacks {
        margin-top: 1rem;
      }

      .feedback:not(:last-child) {
        margin-bottom: 1rem;
      }

      @media all and (min-width: 768px) {
        .feedbackChannel {
          flex-flow: row wrap;
        }

        h2,
        .feedbacks {
          flex: 0 0 100%;
        }

        .toggle {
          flex: 0 0 auto;
        }

        .publicationToggle {
          margin: 0 0 0 2rem;
        }

        .feedback:not(:last-child) {
          margin-bottom: 0.5rem;
        }
      }
    `}</style>
  </div>
)

FeedbackChannel.defaultProps = defaultProps

export default FeedbackChannel
