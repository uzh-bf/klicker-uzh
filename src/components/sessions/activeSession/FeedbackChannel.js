import React from 'react'
import { Checkbox, Header } from 'semantic-ui-react'

import withCSS from '../../../lib/withCSS'

const FeedbackChannel = () => (
  <div>
    <Header dividing as="h2" content="Feedback-Channel" />
    <Checkbox toggle label="Aktiviert" />
    <Checkbox toggle label="Fragen publizieren" />
    <div className="feedbacks">
      Hello
    </div>
    <style jsx>{`
      .feedbacks {
        background: lightgrey;
        margin-top: .5rem;
      }
    `}</style>
  </div>
)

// TODO semantic-ui styling import
export default withCSS(FeedbackChannel, ['checkbox', 'header'])
