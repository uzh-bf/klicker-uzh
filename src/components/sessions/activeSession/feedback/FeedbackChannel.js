import React from 'react'
import PropTypes from 'prop-types'
import { Checkbox } from 'semantic-ui-react'
import { FormattedMessage } from 'react-intl'

import Feedback from './Feedback'

import withCSS from '../../../../lib/withCSS'

const FeedbackChannel = ({ data, intl }) => (
  <div>
    <h2>
      {/* TODO correct naming of identifier */}
      <FormattedMessage
        defaultMessage="Feedback-Channel"
        id="pages.runningSession.feedbackChannel.title"
      />
    </h2>
    <Checkbox
      toggle
      label={intl.formatMessage({
        defaultMessage: 'Aktiviert',
        id: 'pages.runningSession.feedbackChannel.checkbox.activated', // TODO correct naming of identifier
      })}
    />
    <Checkbox
      toggle
      className="publishCheckbox"
      label={intl.formatMessage({
        defaultMessage: 'Fragen publizieren',
        id: 'pages.runningSession.feedbackChannel.checkbox.publishQuestions', // TODO correct naming of identifier
      })}
    />
    <div className="feedbacks">
      {
        data.map(({ content, id, votes }) => <Feedback key={id} content={content} votes={votes} />)
      }
    </div>
    <style jsx>{`
      .feedbacks {
        margin-top: .5rem;
      }
    `}</style>
  </div>
)

FeedbackChannel.propTypes = {
  data: PropTypes.arrayOf({
    content: PropTypes.string,
    id: PropTypes.string,
    votes: PropTypes.number,
  }).isRequired,
  intl: PropTypes.shape({
    formatMessage: PropTypes.func.isRequired,
  }).isRequired,
}

// TODO semantic-ui styling import
export default withCSS(FeedbackChannel, ['checkbox', 'header'])
