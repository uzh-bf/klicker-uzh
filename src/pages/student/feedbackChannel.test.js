/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import FeedbackChannel from './feedbackChannel'

// HACK: workaround for https://github.com/Semantic-Org/Semantic-UI-React/issues/1702
jest.mock('react-dom', () => ({
  findDOMNode: jest.fn(() => {}),
}))

describe('Snapshot-Testing', () => {
  it('Works', () => {
    const component = renderer.create(<FeedbackChannel />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
