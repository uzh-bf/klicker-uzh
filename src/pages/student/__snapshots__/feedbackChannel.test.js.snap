// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot-Testing Works 1`] = `
<div
  className="studentLayout"
  data-jsx={3724175818}
>
  <div
    className="header"
    data-jsx={3724175818}
  >
    <button
      className="ui basic icon button"
      disabled={undefined}
      onClick={[Function]}
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="content icon"
      />
    </button>
    <h1
      data-jsx={3724175818}
    >
      Feedback-Channel
    </h1>
  </div>
  <div
    className="content"
    data-jsx={3724175818}
  >
    <div
      className="sidebar"
      data-jsx={3743778143}
    >
      <div
        className="pushable"
      >
        <div
          className="ui vertical labeled icon ui overlay left wide sidebar sidebarMenu menu"
        >
          <a
            className="item"
            onClick={[Function]}
          >
            <span>
              Active Question
            </span>
          </a>
          <a
            className="active item"
            onClick={[Function]}
          >
            <span>
              Feedback-Channel
            </span>
          </a>
        </div>
        <div
          className="pusher"
        >
          <div
            className="feedbackChannel"
            data-jsx={2117537256}
          >
            <div
              className="confusionArea"
              data-jsx={2117537256}
            >
              <h2
                data-jsx={2117537256}
              >
                Speed
              </h2>
              <div
                className="confusionSlider"
                data-jsx={2117537256}
              >
                <div
                  className="rangeslider rangeslider-horizontal"
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    className="rangeslider__fill"
                    style={
                      Object {
                        "width": "0px",
                      }
                    }
                  />
                  <div
                    className="rangeslider__handle"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    style={
                      Object {
                        "left": "0px",
                      }
                    }
                  >
                    <div
                      className="rangeslider__tooltip"
                    >
                      <span>
                        0
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <h2
                data-jsx={2117537256}
              >
                Difficulty
              </h2>
              <div
                className="confusionSlider"
                data-jsx={2117537256}
              >
                <div
                  className="rangeslider rangeslider-horizontal"
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                >
                  <div
                    className="rangeslider__fill"
                    style={
                      Object {
                        "width": "0px",
                      }
                    }
                  />
                  <div
                    className="rangeslider__handle"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    style={
                      Object {
                        "left": "0px",
                      }
                    }
                  >
                    <div
                      className="rangeslider__tooltip"
                    >
                      <span>
                        0
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="feedbacks"
              data-jsx={2117537256}
            >
              <div
                className="feedback"
                data-jsx={2117537256}
              >
                <div
                  className="feedback"
                  data-jsx={1568002809}
                >
                  <div
                    className="content"
                    data-jsx={1568002809}
                  >
                    hello world
                  </div>
                  <div
                    className="votes"
                    data-jsx={1568002809}
                  >
                    +
                    100
                  </div>
                </div>
              </div>
              <div
                className="feedback"
                data-jsx={2117537256}
              >
                <div
                  className="feedback"
                  data-jsx={1568002809}
                >
                  <div
                    className="content"
                    data-jsx={1568002809}
                  >
                    blablbla
                  </div>
                  <div
                    className="votes"
                    data-jsx={1568002809}
                  >
                    +
                    50
                  </div>
                </div>
              </div>
              <div
                className="feedback"
                data-jsx={2117537256}
              >
                <div
                  className="feedback"
                  data-jsx={1568002809}
                >
                  <div
                    className="content"
                    data-jsx={1568002809}
                  >
                    hello
                  </div>
                  <div
                    className="votes"
                    data-jsx={1568002809}
                  >
                    +
                    25
                  </div>
                </div>
              </div>
              <div
                className="feedback"
                data-jsx={2117537256}
              >
                <div
                  className="feedback"
                  data-jsx={1568002809}
                >
                  <div
                    className="content"
                    data-jsx={1568002809}
                  >
                    hooi
                  </div>
                  <div
                    className="votes"
                    data-jsx={1568002809}
                  >
                    +
                    10
                  </div>
                </div>
              </div>
            </div>
            <div
              className="actionButton"
              data-jsx={2117537256}
            >
              <button
                className="ui large circular icon primary button"
                disabled={undefined}
                onClick={[Function]}
                tabIndex={undefined}
              >
                <i
                  aria-hidden="true"
                  className="plus icon"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
